app/src/main/java/mozilla/lockbox/store/AccountStore.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package mozilla.lockbox.store
+
+import android.content.SharedPreferences
+import io.reactivex.Observable
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.subjects.PublishSubject
+import mozilla.components.service.fxa.FirefoxAccount
+import mozilla.components.service.fxa.OAuthInfo
+import mozilla.components.service.fxa.Profile
+import mozilla.lockbox.support.Optional
+import mozilla.lockbox.support.asOptional
+import mozilla.lockbox.flux.Dispatcher
+import mozilla.lockbox.support.SecurePreferences
+
+private const val FIREFOX_ACCOUNT_KEY = "firefox-account"
+private val FXA_SCOPES = arrayOf("profile", "https://identity.mozilla.com/apps/lockbox")
+
+class AccountStore(
+    private val dispatcher: Dispatcher = Dispatcher.shared,
+    private val securePreferences: SecurePreferences = SecurePreferences()
+) {
+    companion object {
+        val shared = AccountStore()
+    }
+
+    internal val compositeDisposable = CompositeDisposable()
+
+    val oauthInfo: Observable<Optional<OAuthInfo>> = PublishSubject.create()
+    val profile: Observable<Optional<Profile>> = PublishSubject.create()
+
+    init {
+        // registration for further actions happens here
+    }
+
+    fun apply(sharedPreferences: SharedPreferences) {
+        securePreferences.apply(sharedPreferences)
```

Comment:
Pass in the `SecurePreferences` object, not the `SharedPreferences` object.

Reply:
That means `LockboxApplication` will be doing heavier lifting than before, but I suppose that's fine.

Reply:
put me down for "not a fan of this change", unless `SecurePreferences` is maintained/initialized by `LockboxApplication` for other pieces.

Reply:
:sparkles: _no one knows_ :sparkles:

Reply:
I don't like any of the above tbh; I think we agreed to keep the `apply` pattern to get those other PRs landed, but I would rather come up with another solution because there will undoubtedly be many other `Context`-requiring `Store`s.